### ISSUE TITLE
Many unit test failures on localhost

### ISSUE THREAD

**Issue by NovemLinguae** [wikimedia-gadgets/twinkle#1780]
Tests didn't run at all on Jest 27.

On Jest 29, there are a couple of failures and tons of notices. This ticket is to clean up all the failures and all the notices.

```
> test
> jest


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "addOnloadHook" is deprecated. Use jQuery instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as addOnloadHook] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "loadedScripts" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as loadedScripts] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "importScriptURI" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as importScriptURI] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "importStylesheetURI" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as importStylesheetURI] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)

PASS tests/morebits.wikitext.js

  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "addOnloadHook" is deprecated. Use jQuery instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as addOnloadHook] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "loadedScripts" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as loadedScripts] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "importScriptURI" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as importScriptURI] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "importStylesheetURI" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as importStylesheetURI] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)

PASS tests/morebits.date.js
  ● Console

    console.warn
      Invalid Morebits.date initialisation: [ 'no' ]

      1823 | 	// Still no?
      1824 | 	if (!this.isValid()) {
    > 1825 | 		mw.log.warn('Invalid Morebits.date initialisation:', args);
           | 		       ^
      1826 | 	}
      1827 | };
      1828 |

      at new warn (morebits.js:1825:10)
      at Object.<anonymous> (tests/morebits.date.js:21:16)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "addOnloadHook" is deprecated. Use jQuery instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as addOnloadHook] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "loadedScripts" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as loadedScripts] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "importScriptURI" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as importScriptURI] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "importStylesheetURI" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as importStylesheetURI] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)

PASS tests/morebits.string.js

  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "addOnloadHook" is deprecated. Use jQuery instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as addOnloadHook] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "loadedScripts" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as loadedScripts] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "importScriptURI" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as importScriptURI] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "importStylesheetURI" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as importStylesheetURI] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)

FAIL tests/morebits.quickForm.js
  ● quickform › getInputData

    expect(received).toMatchSnapshot()

    Snapshot name: `quickform getInputData 1`

    - Snapshot  - 2
    + Received  + 2

    @@ -1,7 +1,7 @@
    - Object {
    -   "checkboxlist1": Array [
    + {
    +   "checkboxlist1": [
          "checkval2",
        ],
        "checkname1": true,
        "checkname2": false,
        "inputname": "prefilled value",

      112 |
      113 | 	test("getInputData", () => {
    > 114 | 		expect(Morebits.quickForm.getInputData(getRenderedForm())).toMatchSnapshot();
          | 		                                                           ^
      115 | 	});
      116 |
      117 | 	test("getElements", () => {

      at Object.toMatchSnapshot (tests/morebits.quickForm.js:114:62)

  ● quickform › getElements

    expect(received).toMatchSnapshot()

    Snapshot name: `quickform getElements 1`

    - Snapshot  - 1
    + Received  + 1

    @@ -1,6 +1,6 @@
    - Array [
    + [
        <input
          id="node_20_0"
          name="checkboxlist1"
          type="checkbox"
          value="checkval1"

      117 | 	test("getElements", () => {
      118 | 		let form = getRenderedForm();
    > 119 | 		expect(Morebits.quickForm.getElements(form, 'checkboxlist1')).toMatchSnapshot();
          | 		                                                              ^
      120 | 	});
      121 |
      122 | });

      at Object.toMatchSnapshot (tests/morebits.quickForm.js:119:65)

 › 2 snapshots failed.

  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "addOnloadHook" is deprecated. Use jQuery instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as addOnloadHook] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "loadedScripts" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as loadedScripts] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "importScriptURI" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as importScriptURI] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "importStylesheetURI" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as importStylesheetURI] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)

PASS tests/twinklewarn.test.js

  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "addOnloadHook" is deprecated. Use jQuery instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as addOnloadHook] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "loadedScripts" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as loadedScripts] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "importScriptURI" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as importScriptURI] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "importStylesheetURI" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as importStylesheetURI] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)

PASS tests/morebits.ip.js

  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "addOnloadHook" is deprecated. Use jQuery instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as addOnloadHook] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "loadedScripts" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as loadedScripts] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "importScriptURI" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as importScriptURI] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "importStylesheetURI" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as importStylesheetURI] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)

PASS tests/morebits.createHtml.js

  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "addOnloadHook" is deprecated. Use jQuery instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as addOnloadHook] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "loadedScripts" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as loadedScripts] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "importScriptURI" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as importScriptURI] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "importStylesheetURI" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as importStylesheetURI] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)

PASS tests/morebits.js

  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "addOnloadHook" is deprecated. Use jQuery instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as addOnloadHook] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "loadedScripts" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as loadedScripts] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "importScriptURI" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as importScriptURI] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "importStylesheetURI" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as importStylesheetURI] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)

PASS tests/morebits.taskManager.js
  ● Console

    console.error
      Error: Error: connect ECONNREFUSED ::1:80
          at Object.dispatchError (/home/runner/work/twinkle/twinkle/node_modules/jsdom/lib/jsdom/living/xhr/xhr-utils.js:63:19)
          at Request.<anonymous> (/home/runner/work/twinkle/twinkle/node_modules/jsdom/lib/jsdom/living/xhr/XMLHttpRequest-impl.js:655:18)
          at Request.emit (node:events:525:35)
          at ClientRequest.<anonymous> (/home/runner/work/twinkle/twinkle/node_modules/jsdom/lib/jsdom/living/helpers/http-request.js:121:14)
          at ClientRequest.emit (node:events:513:28)
          at Socket.socketErrorListener (node:_http_client:502:9)
          at Socket.emit (node:events:513:28)
          at emitErrorNT (node:internal/streams/destroy:151:8)
          at emitErrorCloseNT (node:internal/streams/destroy:116:3)
          at processTicksAndRejections (node:internal/process/task_queues:82:21) {
        type: 'XMLHttpRequest'
      }

      at VirtualConsole.<anonymous> (node_modules/jest-environment-jsdom/build/index.js:60:23)
      at Object.dispatchError (node_modules/jsdom/lib/jsdom/living/xhr/xhr-utils.js:66:53)
      at Request.<anonymous> (node_modules/jsdom/lib/jsdom/living/xhr/XMLHttpRequest-impl.js:655:18)
      at ClientRequest.<anonymous> (node_modules/jsdom/lib/jsdom/living/helpers/http-request.js:121:14)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "addOnloadHook" is deprecated. Use jQuery instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as addOnloadHook] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "loadedScripts" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as loadedScripts] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "importScriptURI" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as importScriptURI] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "importStylesheetURI" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as importStylesheetURI] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)

PASS tests/morebits.unbinder.js

  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "addOnloadHook" is deprecated. Use jQuery instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as addOnloadHook] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "loadedScripts" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as loadedScripts] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "importScriptURI" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as importScriptURI] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "importStylesheetURI" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as importStylesheetURI] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)

PASS tests/morebits.array.js

  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "addOnloadHook" is deprecated. Use jQuery instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as addOnloadHook] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "loadedScripts" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as loadedScripts] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "importScriptURI" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as importScriptURI] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Use of "importStylesheetURI" is deprecated. Use mw.loader instead.".

      at console.warn (node_modules/@jest/console/build/BufferedConsole.js:191:10)
      at maybeLog (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:52:11)
      at Window.get [as importStylesheetURI] (node_modules/mock-mediawiki/lib/mediawiki.base/log.js:68:5)
          at Array.forEach (<anonymous>)

PASS tests/morebits.batchOperation.js

Snapshot Summary
 › 2 snapshots failed from 1 test suite. Inspect your code changes or run `npm test -- -u` to update them.

Test Suites: 1 failed, 11 passed, 12 total
Tests:       2 failed, 53 passed, 55 total
Snapshots:   2 failed, 7 passed, 9 total
Time:        3.682 s
Ran all test suites.
##[error]Process completed with exit code 1.

**Comment by NovemLinguae** [comment]
Looks like the failures are of 3 types:

* [ ] Snapshot tests failing
* [ ] Async/await problems
* [ ] Deprecated functions being called in the library mock-mediawiki. Will probably need to be addressed upstream via wikimedia-gadgets/mock-mediawiki#1

**Comment by siddharthvp** [comment]
Almost all of these deprecation warnings for interfaces that have been undeprecated in more recent versions of MediaWiki than the one mock-mediawiki is using.

**Comment by siddharthvp** [comment]
Have released v1.4.0 of mock-mediawiki which removes most of the notices. The one on the `mw.log.warn` line in morebits.date test can be prevented by stubbing out the logger if we wish. 

I do get this one error on morebits.taskManager test though:
```
    console.error
      Error: Error: connect ECONNREFUSED ::1:80
          at Object.dispatchError (/Users/siddharthvp/bashhome/twinkle/node_modules/jsdom/lib/jsdom/living/xhr/xhr-utils.js:63:19)
          at Request.<anonymous> (/Users/siddharthvp/bashhome/twinkle/node_modules/jsdom/lib/jsdom/living/xhr/XMLHttpRequest-impl.js:655:18)
          at Request.emit (node:events:524:35)
          at ClientRequest.<anonymous> (/Users/siddharthvp/bashhome/twinkle/node_modules/jsdom/lib/jsdom/living/helpers/http-request.js:121:14)
          at ClientRequest.emit (node:events:512:28)
          at Socket.socketErrorListener (node:_http_client:495:9)
          at Socket.emit (node:events:512:28)
          at emitErrorNT (node:internal/streams/destroy:151:8)
          at emitErrorCloseNT (node:internal/streams/destroy:116:3)
          at processTicksAndRejections (node:internal/process/task_queues:82:21) {
        type: 'XMLHttpRequest'
      }
```
which seems peculiar as the test doesn't do any HTTP requests at all.
